### PR TITLE
T1 and T2star mapping

### PR DESCRIPTION
**Pull request by ckolbPTB**

### CODE REVIEW

**Comment by ckolbPTB** [comment]
Mypy errors:
examples/qmri_sg_challenge_2024_t2_star.py:55: error: Incompatible types in assignment (expression has type "Data", variable has type "IData")  [assignment]

examples/qmri_sg_challenge_2024_t1.py:68: error: Unsupported operand types for @ ("MagnitudeOp" and "InversionRecovery")  [operator]

examples/qmri_sg_challenge_2024_t1.py:82: error: Unsupported operand types for @ ("MSEDataDiscrepancy" and "Operator[Tensor, Tensor, tuple[Tensor, ...]]")  [operator]

**Comment by fzimmermann89** [comment]
If we want to merge this before ismrm, it would be fine with me.

There are a couple of issues, related to typing of mse_data_consisntency, magntitude_op and phase_op.
But these can be tracked via #274 and the functional implementation.

Are we ok with having the type fixes in this PR? Or should we separate them out? 

Again, if the goal is to get this merged before ISMRM I would be fine either way.

**Review by fzimmermann89 — APPROVED** [review]
if we want to merge before ismrm, it is ok.
otherwise, we should consider moving the typing fixes into a separate PR (and fix all issues in  one PR)

**Comment by ckolbPTB** [comment]
> Are we ok with having the type fixes in this PR? Or should we separate them out?

It would be great if we could do the type fixes in a separate PR. Happy to wait with merging this one.

**Comment by fzimmermann89** [comment]
So, this will have to wait with merging until the stack #281 , #279 , #282 is merged

**Review by fzimmermann89 — CHANGES_REQUESTED** [review]
Just blocking it for a sane order of merges, I will rebase it after the changes I split out of this PR are merged.

**Comment by ckolbPTB** [comment]
Before we merge this one, we should also merge #271 and then adapt these lines:

https://github.com/PTB-MR/mrpro/blob/34f80bf80395d478699fe1051c9b0126e62cf56b/examples/qmri_sg_challenge_2024_t1.py#L68

https://github.com/PTB-MR/mrpro/blob/34f80bf80395d478699fe1051c9b0126e62cf56b/examples/qmri_sg_challenge_2024_t2_star.py#L74

**Review by schuenke — APPROVED** [review]
I reverted all changes in the Operator files and did some typos, spelling, punctuation and formatting corrections in the scripts. Hope it's all consistent now.

**Comment by schuenke** [comment]
> Just blocking it for a sane order of merges, I will rebase it after the changes I split out of this PR are merged.

#279, #281 and #282 are merged, so we should be fine merging this one, right @fzimmermann89 ?

File: .vscode/settings.json [hunk 1]
```diff
@@ -18,4 +18,5 @@
     ],
     "python.testing.unittestEnabled": false,
     "python.testing.pytestEnabled": true,
+    "restructuredtext.pythonRecommendation.disabled": true,
```
Comment (fzimmermann89) [inline .vscode/settings.json]: what does this do?

**Review by fzimmermann89 — APPROVED** [review]
at some point, we should think about merging the L2 functional and the Msedatadiscrepency

File: .vscode/settings.json (hunk 1 above)
Comment (ckolbPTB) [reply .vscode/settings.json]: not sure anymore, can probably be removed...

**Pull request merged**